Automattic/wp-calypso
# Posts: Remove lib/post-metadata

#### Changes proposed in this Pull Request

Many of the functions in `/lib/post-metadata` are unused. The rest are just used once, and are very simple: they retrieve a meta value by a key that matches certain conditions. Once we remove the unused, we can move the remaining ones to the respective components without repetition, and that would allow us to avoid unnecessarily sharing code between the consumer components.

#### Testing instructions

* Go to `/devdocs/blocks/post-share` and verify it still works as it did before.
* Go to `/posts/:site`
  * Click on the dots of a post, click "Share" and verify the `PostShare` component there still works like before.
  * Click on the dots of a post, click "View", select the "Search and Social" option from the dropdown, and go through all previews (Google search, Facebook, Twitter) and verify they still work like before.
